src/contexts/price-oracle.context.tsx

```diff
@@ -32,7 +50,19 @@ const PriceOracleProvider: FC = (props) => {
       throw new Error("Price oracle contract is not available");
     }
 
-    return priceOracleContract.getRate(tokenAddress, env.REACT_APP_USDT_CONTRACT_ADDRESS, true);
+    if (fiatExchangeRates === undefined) {
+      throw new Error("Fiat exchange rates are not available");
+    }
+
+    const rate = await priceOracleContract.getRate(
+      tokenAddress,
+      env.REACT_APP_USDT_CONTRACT_ADDRESS,
+      true
+    );
+    const usdPrice = Number(formatUnits(rate, USDT_DECIMALS));
+    const fiatExchangeRate = fiatExchangeRates[preferredCurrency];
```

Comment:
This could be undefined

Reply:
Done! I prefer the `Partial<Record<keyof typeof Currency, number>>` approach